### PR TITLE
Added more benchmarks for linear algebra

### PR DESCRIPTION
**Pull request by kshyatt**
`eig` for special matrix types and a bunch of BLAS.

### CODE REVIEW

File: src/lapack/LAPACKBenchmarks.jl [hunk 1]
```diff
@@ -17,11 +17,31 @@ const SIZES = (2^2, 2^4, 2^6, 2^8, 2^10)
...
         [(:realeig, size(A, 1)) => eig(A) for A in real_arrays]
         [(:symeig, size(A, 1)) => eig(A) for A in sym_arrays]
         [(:hermeig, size(A, 1)) => eig(A) for A in herm_arrays]
+        [(:symtrieig, size(A, 1)) => eigfact(A) for A in symtri_arrays]
+        [(:uptrieig, size(A, 1)) => eigfact(A) for A in uptri_arrays]
+        [(:lotrieig, size(A, 1)) => eigfact(A) for A in lotri_arrays]
+        [(:unituptrieig, size(A, 1)) => eigfact(A) for A in unituptri_arrays]
+        [(:unitlotrieig, size(A, 1)) => eigfact(A) for A in unitlotri_arrays]
     end
     @tags "array" "lapack" "linalg" "eig"
```
Comment (jrevels) [inline src/lapack/LAPACKBenchmarks.jl]: Awesome, thanks for this! Can you add the `"eigfact"` and `"factorization"` tags to this? Otherwise, LGTM.

**Comment by kshyatt** [comment]
Done! Tags added :)

**Comment by kshyatt** [comment]
Since the tags are now there, and I got the go ahead on the other stuff, I'm going to merge!

**Comment by jrevels** [comment]
:+1:

**Pull request merged**